portalnet/src/overlay/protocol.rs

```diff
@@ -509,13 +509,10 @@ where
     pub async fn send_offer(
         &self,
         enr: Enr,
-        content_key: RawContentKey,
-        content_value: Vec<u8>,
+        content_items: Vec<(RawContentKey, Vec<u8>)>,
     ) -> Result<Accept, OverlayRequestError> {
         // Construct the request.
-        let request = Request::PopulatedOffer(PopulatedOffer {
-            content_items: vec![(content_key, content_value)],
-        });
+        let request = Request::PopulatedOffer(PopulatedOffer { content_items });
```

Comment:
In my opinion, check that number of content items is 1..=64 should happen here (in which case it should return `Err(OverlayRequestError::InvalidRequest(..))`).

In that case, it's probably not needed in `*_rpc.rs`.

Reply:
I don't think we should move the request check to `portalnet/src/overlay/protocol.rs`. Because then we would be sending possibly over 64 key-value pairs through the `*_rx` channel which I think should be avoided. Yes this suggestion would reduce code duplication, but I don't think we should push validating length checks after a channel. We should do the range check as soon as possible to avoid that send. Why waste extra resources when it can be avoided.

Reply:
The reason why I think it should be there is not because I want to avoid code duplication, but because this type of check should be at lower level. Reason being is that what if there is alternative call to this function (or is added in the future).

Now that I think about it, we might want it at even lower level (e.g. right before sending offer request over network). Because poke, and gossip on deletion (and maybe some other mechanics) wouldn't have this check.

But, regarding your point, it's fine to have it on upper level as well (and not make calls that we know would fail).

Reply:
We do have a check at the lower level https://github.com/ethereum/trin/blob/master/portalnet/src/gossip.rs#L103-L115 ... but i'd include an update in this pr to use the `MAX_AMOUNT_OF_OFFERED_CONTENT_KEYS` const in that code

Reply:
I will update the code to use that const. I think it is fine if we have a lower and higher level check